java-client/src/test/java/org/opensearch/client/opensearch/indices/AliasDefinitionTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.client.opensearch.indices;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+
+import org.junit.Test;
+import org.opensearch.client.opensearch._types.query_dsl.Query;
+import org.opensearch.client.opensearch.model.ModelTestCase;
+
+public class AliasDefinitionTest extends ModelTestCase {
+
+    @Test
+    public void toBuilder() {
+        AliasDefinition origin = new AliasDefinition.Builder().filter(f -> f.wrapper(w -> w.query("aaa"))).build();
```

Comment:
These tests could be simplified a bit by just applying to the `Query` class

Reply:
We have a bunch of tests for query but those did not actually catch the issue.

Reply:
It needs to specifically be wrapped within an instance of the `Query` class (ie. testing on `Query.of(q -> q.wrapper(w -> w.query("eyJib29sIiA6IHsibXVzdCIgOiB7InRlcm0iIDogeyJ0eXBlIiA6ICJhYmMifX19fQo")))`) as the issue was `Query` missing the deserialize handler for `"wrapper"`.

WrapperQuery on its own:
```json
{
  "query": "eyJib29sIiA6IHsibXVzdCIgOiB7InRlcm0iIDogeyJ0eXBlIiA6ICJhYmMifX19fQo"
}
```

Query(WrapperQuery):
```json
{
  "wrapper": {
    "query": "eyJib29sIiA6IHsibXVzdCIgOiB7InRlcm0iIDogeyJ0eXBlIiA6ICJhYmMifX19fQo"
  }
}
```

Reply:
> It needs to specifically be wrapped within an instance of the `Query` class (ie. testing on `Query.of(q -> q.wrapper(w -> w.query("eyJib29sIiA6IHsibXVzdCIgOiB7InRlcm0iIDogeyJ0eXBlIiA6ICJhYmMifX19fQo")))`) as the issue was `Query` missing the deserialize handler for `"wrapper"`.

Correct, the query tests never exercise the deserialization flows (afaict)